### ISSUE TITLE
License issue?

### ISSUE THREAD

**Issue by riccardobl** [voidqk/sndfilter#7]
Hello, first of all, nice work. :squirrel:
I'm working on an opensource code that will be released under a permissive license and i have add some reverberation to an audio source, your reverb implementation is exactly what im looking for, but i've read the readme and it says

> The reverb effect is a complete rewrite of Freeverb3's Progenitor2 algorithm. 

afaik Freeverb3 is a gnu library under gnu gpl license, due to the properties of this license, shouldn't this code be licensed under gpl aswell?

**Comment by KaneRoot** [comment]
Maybe is there a way to isolate the GPL license to this file only. I just discovered this project yesterday and it's awesome, hope to integrate it in my projects some day, don't want it to be GPL'd.

**Comment by voidqk** [comment]
I don't believe it falls under GPL, because it is a rewrite from scratch.  I.e., no copy/pasted code, and if you actually look at Freeverb vs sndfilter code, you can tell it is completely different.  The algorithm is the same (or, should be), but the implementation is completely different.

This is no different than Linux being a rewrite of Unix, imo.

Of course, I'm not a lawyer.  But I can tell you that I spent a lot of effort reading the Freeverb codebase, figuring out the algorithm behind it, and reimplementing it.  Like I said, just compare the source files, it's very obvious.

**Comment by riccardobl** [comment]
Thanks for the reply.
It seems to me that this may fall into a grey area  :thinking: , since gpl covers derivate work, and this may be seen as a derivate work, or at least that's how the last part in the readme may be interpreted, but I'm not a lawyer either.

**Comment by weepy** [comment]
generally it's not possible to license an algorithm
